### PR TITLE
Enable the isort rules for ruff and fix linting issues

### LINKED ISSUE #3695: Ruff failing to correctly sort imports
## Description

This might be a regression caused by #3546, which replaced isort with Ruff. The pre-commit step is no longer correctly fixing the order of imports.

## Reproduction

1. Choose any file, for example `openverse/catalog/dags/providers/provider_workflows.py`. Move the existing imports around so they are out of order: for example, move some imports within an import block so they are not alphabetical, and also move some imports to the wrong block (ie local imports before third party).
2. Run the pre-commit and observe that it passes and makes no changes.

### PR DESCRIPTION
**Pull request by stacimc**
## Fixes

Fixes #3695 by @stacimc 

## Description

When we moved from isort to ruff, the import sorting in the pre-commit step was broken. This PR:

* Updates `pyproject.toml` to enable the isort rules for ruff, which causes import sorting to begin working again in the pre-commit.
* Fixes all linting issues with imports that were merged while the pre-commit wasn't working.

## Testing Instructions

Pick any file and move the imports out of order (out of alphabetical order, change the spaces between import blocks, move local imports before third party, etc). Run `just lint` and verify it fails and updates the file. Verify that no other files were changed (all other fixes should be present in this PR).

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by AetherUnbound** [comment]
Ah, it looks like we can get `elasticsearch_cluster` to show up properly by adding it here!

https://github.com/WordPress/openverse/blob/ba942f21c6d9922c64aa8d4bb0b8b4b8311cfe27/pyproject.toml#L41-L52

I had forgotten that we needed to make those explicit due to how our python path is set up on Airflow 😅

**Review by AetherUnbound — APPROVED** [review]
Awesome find!!

File: pyproject.toml [hunk 1]
```diff
@@ -45,6 +46,7 @@ known-first-party = [
   "common",
   "data_refresh",
   "database",
+  "elasticsearch_cluster",
   "maintenance",
   "oauth2",
   "providers",
```
Comment (dhruvkb) [inline pyproject.toml:52]: Please also include `"test"` here. It's a first party in the API and the ingestion server. See #3700 for more info.

File: pyproject.toml [hunk 2]
```diff
@@ -12,6 +12,7 @@ ignore-words = ".codespell/ignore_words.txt"
 extend-select = [
   "UP",
```
Comment (dhruvkb) [inline pyproject.toml:13]:
```suggestion
  "UP",  # pyupgrade, https://docs.astral.sh/ruff/rules/#pyupgrade-up
```

File: pyproject.toml [hunk 3]
```diff
@@ -12,6 +12,7 @@ ignore-words = ".codespell/ignore_words.txt"
...
   "D",  # pydocstyle, https://docs.astral.sh/ruff/rules/#pydocstyle-d
+  "I",  # isort
```
Comment (dhruvkb) [inline pyproject.toml:15]:
```suggestion
  "I",  # isort, https://docs.astral.sh/ruff/rules/#isort-i
```

**Review by dhruvkb — CHANGES_REQUESTED** [review]
Good catch, I have some additional notes. Thanks!

File: pyproject.toml (hunk 1 above)
Comment (stacimc) [reply pyproject.toml:52]: Great catch!

**Review by dhruvkb — APPROVED** [review]

**Pull request merged**